flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerDriverTest.java

```diff
@@ -669,4 +650,116 @@ void prepareForTestStartTaskExecutorProcessVariousSpec(
+                        ContainerState.COMPLETE,
+                        "disk failed exit code",
+                        ContainerExitStatus.DISKS_FAILED);
+        testingGetConatainerCompletedCause(containerStatus);
+    }
+
+    @Test
+    public void testGetConatainerCompletedCauseForUnknown() throws Exception {
+        ContainerStatus containerStatus =
+                ContainerStatusPBImpl.newInstance(
+                        testingContainer.getId(), ContainerState.COMPLETE, "unknown exit code", -1);
+        testingGetConatainerCompletedCause(containerStatus);
+    }
+
+    public void testingGetConatainerCompletedCause(ContainerStatus containerStatus)
+            throws Exception {
+        new Context() {
+            {
+                addContainerRequestFutures.add(new CompletableFuture<>());
+                addContainerRequestFutures.add(new CompletableFuture<>());
+
+                testingYarnAMRMClientAsyncBuilder.setAddContainerRequestConsumer(
+                        (ignored1, ignored2) ->
+                                addContainerRequestFutures
+                                        .get(
+                                                addContainerRequestFuturesNumCompleted
+                                                        .getAndIncrement())
+                                        .complete(null));
+                resourceEventHandlerBuilder.setOnWorkerTerminatedConsumer(
+                        (ignore1, ignore2) ->
+                                getDriver().requestResource(testingTaskExecutorProcessSpec));
+
+                runTest(
+                        () -> {
+                            runInMainThread(
+                                    () ->
+                                            getDriver()
+                                                    .requestResource(
+                                                            testingTaskExecutorProcessSpec));
+                            resourceManagerClientCallbackHandler.onContainersAllocated(
+                                    ImmutableList.of(testingContainer));
+                            ContainerStatus testingContainerStatus =
+                                    createTestingContainerCompletedStatus(containerStatus);
+
+                            ResourceManagerDriver resourceManagerDriver = getDriver();
+                            if (resourceManagerDriver instanceof YarnResourceManagerDriver) {
+                                YarnResourceManagerDriver yarnResourceManagerDriver =
+                                        (YarnResourceManagerDriver) resourceManagerDriver;
+                                assertTrue(
+                                        yarnResourceManagerDriver
+                                                .getConatainerCompletedCause(testingContainerStatus)
+                                                .contains(testingContainerStatus.getDiagnostics()));
```

Comment:
I am wondering whether we can't test this in isolation if `getConatainerCompletedCause` is a static utility method. Independent of this change, I think it is a good improvement that we test the overall `onContainersCompleted` method for the different `ContainerExitStatus`.